File: Reconocedor de expresiones matemáticas/gofuncs/math_2.go
package main

import (
	"C"
	"math"
)

//export cbrtF
func cbrtF(x float64) float64 {
	return math.Cbrt(x)
}

//export factorial
func factorial(x float64) float64 {
	return x * math.Gamma(x)
}

//export sqrtF
func sqrtF(x float64) float64 {
	return math.Sqrt(x)
}

//export gammaF
func gammaF(x float64) float64 {
	return math.Gamma(x)
}

//export log2F
func log2F(x float64) float64 {
	return math.Log2(x)
}

//export acoshF
func acoshF(x float64) float64 {
	return math.Acosh(x)
}

//export asenhF
func asenhF(x float64) float64 {
	return math.Asinh(x)
}

//export atanhF
func atanhF(x float64) float64 {
	return math.Atanh(x)
}

//export hypotF
func hypotF(x, y float64) float64 {
	return math.Hypot(x, y)
}

func main() {}

File: math_2.c
#include <math.h>

double cbrtF(double x) { return cbrt(x); }
double sqrtF(double x) { return sqrt(x); }
double log2F(double x) { return log2(x); }
double acoshF(double x) { return acosh(x); }
double asenhF(double x) { return asinh(x); }
double atanhF(double x) { return atanh(x); }
double hypotF(double x, double y) { return hypot(x, y); }

// La función Gamma en C está en math.h como tgamma
double gammaF(double x) { return tgamma(x); }

// Factorial usando Gamma: x! = gamma(x + 1)
double factorial(double x) { 
    return tgamma(x + 1.0); 
}
